pkg/services/sqlstore/migrations/ualert/alert_rule.go

```diff
@@ -206,6 +207,56 @@ func fixGraphiteReferencedSubQueries(queryData map[string]json.RawMessage) map[s
 	return queryData
 }
 
+// fixPrometheusBothTypeQuery converts Prometheus 'Both' type queries to range queries.
+func fixPrometheusBothTypeQuery(l log.Logger, queryData map[string]json.RawMessage) map[string]json.RawMessage {
+	// There is the possibility to support this functionality by:
+	//	- Splitting the query into two: one for instant and one for range.
+	//  - Splitting the condition into two: one for each query, separated by OR.
+	// However, relying on a 'Both' query instead of multiple conditions to do this in legacy is likely
+	// to be unintentional. In addition, this would require more robust operator precedence in classic conditions.
+	// Given these reasons, we opt to convert them to range queries and log a warning.
+
+	// Check if the query is for Prometheus.
+	ds, ok := queryData["datasource"]
+	if !ok {
+		return queryData
```

Comment:
Let's leave a debug message if datasource field is not found. IIRC, this is an optional field on the SSE backend (and probably anywhere else) and it does not rely on it.

Reply:
What information would the debug message show? If the `datasource` is not required, the only thing we know is that the rule could be prometheus, and could be `Both` and if it is, it won't convert correctly.

Reply:
I could change up the order, try to find `instant` and `range` first. Then if they are both true and the `datasource` is missing, I can log a more informative message. WDYT?

Reply:
> I could change up the order, try to find `instant` and `range` first. Then if they are both true and the `datasource` is missing, I can log a more informative message. WDYT?

I like the idea.

Reply:
Fixed: b3bc9aadf91d9b7b6548728beaa8e1ab8fe1f143